jax/interpreters/masking.py

```diff
@@ -213,16 +229,16 @@ def is_constant(self):
 
 class Mon(dict):
   def __hash__(self):
-    return hash(tuple(self.items()))
+    return hash(tuple(sorted(self.items())))
```

Comment:
I notice the same is true of `Poly.__hash__`/`Mon.__lt__` (i.e. they use sorted + lt instead of frozenset(items)), this would make sense if it was possible to get repeated items but I don't think that is possible, so maybe we should change that too? Apologies if I've got this wrong...

Reply:
I see now that `sorted` is also used for building string representations, and that makes sense I think, so I guess the thing to do is leave the `__lt__` methods there, but use `frozenset` for hashing.